src/main/scala/com/ironcorelabs/IronSdk.scala

```diff
@@ -45,25 +45,135 @@ trait IronSdk[F[_]] {
     documentDecrypt(ByteVector.view(encryptedBytes))
 
   /**
-   * Creates a group. The creating user will become a group admin.
+   * List all of the documents that the current user is able to decrypt.
    *
-   * @param options group creation options. Use `new GroupCreateOpts()` for defaults
+   * @return a List of [[DocumentListMeta]] with metadata about each document the user can decrypt.
+   */
+  def documentList: F[List[DocumentListMeta]]
+
+  /**
+   * Get the metadata for a specific document given its id.
+   *
+   * @param id unique id of the document to retrieve
+   * @return [[DocumentMetadataResult]] with details about the requested document.
+   */
+  def documentGetMetadata(id: DocumentId): F[DocumentMetadataResult]
+
+  /**
+   * Attempt to parse the document id out of an encrypted document.
+   *
+   * @param encryptedDocument encrypted document bytes
+   * @return extracted id
+   */
+  def documentGetIdFromBytes(encryptedDocument: ByteVector): F[DocumentId]
+
+  /**
+   * Attempt to parse the document id out of an encrypted document.
+   *
+   * @param encryptedDocument encrypted document bytes
+   * @return extracted id
+   */
+  def documentGetIdFromBytes(encryptedDocument: Array[Byte]): F[DocumentId] =
+    documentGetIdFromBytes(ByteVector.view(encryptedDocument))
+
+  /**
+   * Update a document name to a new value or clear its value.
+   *
+   * @param id      id of the document to update
+   * @param name    new name for the document. Provide a Some([[DocumentName]]) to update to a new name or None to clear the name field.
+   * @return metadata about the document that was updated.
+   */
+  def documentUpdateName(id: DocumentId, name: Option[DocumentName]): F[DocumentMetadataResult]
+
+  /**
+   * Update the encrypted content of an existing document. Persists any existing access to other users and groups.
+   *
+   * @param id               id of document to update
+   * @param newDocumentData  updated document content to encrypt
+   */
+  def documentUpdateBytes(id: DocumentId, newDocumentData: ByteVector): F[DocumentEncryptResult]
+
+  /**
+   * Update the encrypted content of an existing document. Persists any existing access to other users and groups.
+   *
+   * @param id               id of document to update
+   * @param newDocumentData  updated document content to encrypt
+   */
+  def documentUpdateBytes(id: DocumentId, newDocumentData: Array[Byte]): F[DocumentEncryptResult] =
+    documentUpdateBytes(id, ByteVector.view(newDocumentData))
+
+  /**
+   * Grant access to a document. Recipients of document access can be either users or groups.
+   *
+   * @param documentId id of the document whose access is being modified
+   * @param userGrants list of user grants
+   * @param groupGrants list of group grants
+   * @return information about which grants succeeded/failed
+   */
+  def documentGrantAccess(
+    documentId: DocumentId,
+    userGrants: List[UserId],
```

Comment:
I don't have a strong feeling for which one to be more consistent with, but since we have to call the java method which has them broken up, it's lower complexity to have it the same here.

Reply:
I don't care too much either, honestly. We do use UserOrGroupId in other places, so we just need to convince ourselves that we're at least being internally consistent within ironoxide-scala